src/test/java/test/retryAnalyzer/ExitCodeTest.java

```diff
@@ -25,7 +26,23 @@ public void exitsWithNonzeroOnFailure() {
   @Test
   public void exitsWithZeroAfterSuccessfulRetry() {
     TestNG tng = create(EventualSuccess.class);
+    tng.addListener((ITestNGListener) new TestResultPruner());
     tng.run();
     assertEquals(tng.getStatus(), 0);
   }
+
+  @Test(description = "GITHUB-217")
+  public void exitWithNonzeroOnSkips() {
+    TestNG tng = create(Issue217TestClassSample.class);
+    tng.run();
+    assertEquals(tng.getStatus(), 5);
```

Comment:
5 means HAS_FAILURE and HAS_FSP, right?

Reply:
Because `tng. hasFailureWithinSuccessPercentage()` and `tng. hasFailure()` will be both `true`.

Reply:
Please see this table : https://github.com/cbeust/testng/pull/1579#discussion_r145679187